test-dts/defineComponent.test-d.tsx

```diff
@@ -39,6 +39,9 @@ describe('with object props', () => {
     kkk?: any
     validated?: string
     date?: Date
+    customTypedValidated: 1
+    customTypedValidated2?: string | 1
+    customTypedValidated3?: number
```

Comment:
I am not very happy with the type after this inference.I would like to infer it as [1] here, but [number] will overwrite [1]. And with the existing type structure, I haven't thought of a solution. In other words, if the type of [type] is now a superset of [T] (for xx is T), then [type] will cover [T]

Reply:
I don't think we can solve it with the existing type structure and in order not to cause semantic changes in the existing code.